imap_processing/codice/l0/decompress_codice.py

```diff
@@ -0,0 +1,111 @@
+"""
+
+import lzma
+
+from . import LOSSY_A_TABLE, LOSSY_B_TABLE
+
+
+class CodiceDecompression:
+    """Main class that contains implementations of the decompression algorithms.
+
+    When an algorithm is applied, the ``decompressed_value`` attribute is set or
+    updated to contain the 24- or 32-bit decompressed value.
+
+    Attributes
+    ----------
+    compressed_value : int
+        The 8-bit compressed value to decompress
+    algorithm : str
+        The algorithm to apply. Supported algorithms include 'no compression',
+        'lossyA', 'lossyB', 'lossless', 'lossyA+lossless', and 'lossyB+lossless'
+    decompressed_value : int or None
+        The 24- or 32-bit decompressed value, or None if the algorithm is not
+        supported or hasn't been applied yet.
+    """
+
+    def __init__(self, compressed_value, algorithm):
+        self.compressed_value = compressed_value
+        self.algorithm = algorithm
+        self.decompressed_value = None
+
+    def _apply_lossy_a(self, value):
+        """Apply 8-bit to 32-bit Lossy A decompression algorithm.
+
+        The Lossy A algorithm uses a lookup table imported into this module.
+
+        Parameters
+        ----------
+        value : int
+            The compressed 8-bit value
+        """
+        self.decompressed_value = LOSSY_A_TABLE[value]
+
+    def _apply_lossy_b(self, value):
+        """Apply 8-bit to 32-bit Lossy B decompression algorithm.
+
+        The Lossy B algorithm uses a lookup table imported into this module.
+
+        Parameters
+        ----------
+        value : int
+            The compressed 8-bit value
+        """
+        self.decompressed_value = LOSSY_B_TABLE[value]
+
+    def _apply_lzma_lossless(self, value):
+        """Apply LZMA lossless decompression algorithm.
+
+        Parameters
+        ----------
+        value : int
+            The compressed 8-bit value
+        """
+        self.decompressed_value = lzma.decompress(value)
+        self.decompressed_value = int.from_bytes(
+            self.decompressed_value, byteorder="big"
+        )
+
+    def decompress(self):
+        """Decompress the value.
+
+        Apply the appropriate decompression algorithm(s) based on the value
+        of the ``algorithm`` attribute. One or more individual algorithms may be
+        applied to a given compressed value.
+        """
+        if self.algorithm == "no compression":
+            self.decompressed_value = self.compressed_value
+        elif self.algorithm == "lossyA":
+            self._apply_lossy_a(self.compressed_value)
+        elif self.algorithm == "lossyB":
+            self._apply_lossy_b(self.compressed_value)
+        elif self.algorithm == "lossless":
+            self._apply_lzma_lossless(self.compressed_value)
+        elif self.algorithm == "lossyA+lossless":
+            self._apply_lzma_lossless(self.compressed_value)
+            self._apply_lossy_a(self.decompressed_value)
```

Comment:
The order of these confused me at first until I read the class docstring above indicating that these are the inverse of the original algorithm that did the compressing. Thank you for adding the great documentation here!